### PR TITLE
Update SDL to SDL2

### PR DESCRIPTION
**Pull request by Lunanne**
I don't know whether there is any interest in PRs on this project but I felt like working on it.

### CODE REVIEW

File: src/Main.cpp [hunk 1]
```diff
@@ -26,15 +28,16 @@
...
 PFNGLCOMPRESSEDTEXIMAGE2DARBPROC GLX_CompressedTexImage2D = 0;
 
 #ifdef WIN32
-  int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance,
-                     LPSTR lpCmdLine, int nCmdShow)
+int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance,
+                   LPSTR lpCmdLine, int nCmdShow)
 #else
-  int main(int argc, char* argv[])
+
```
Comment (zrax) [inline src/Main.cpp]: Unnecessary newline

File: src/Main.cpp [hunk 2]
```diff
@@ -58,7 +61,7 @@ PFNGLCOMPRESSEDTEXIMAGE2DARBPROC GLX_CompressedTexImage2D = 0;
         while (!settings.eof()) {
             imString ln = settings.readLine();
             size_t commentPos = ln.find('#');
-            if (commentPos != (size_t)-1)
+            if (commentPos != (size_t) -1)
```
Comment (zrax) [inline src/Main.cpp]: If you're going to change the formatting here, I'd rather see it converted to a C++ style cast (`static_cast<size_t>(-1)`)

File: src/Main.cpp [hunk 3]
```diff
@@ -73,7 +76,7 @@ PFNGLCOMPRESSEDTEXIMAGE2DARBPROC GLX_CompressedTexImage2D = 0;
                 winHeight = strtol(ln.data(), NULL, 0);
             } else {
                 size_t cmdPos = ln.find(' ');
-                if (cmdPos != (size_t)-1)
+                if (cmdPos != (size_t) -1)
```
Comment (zrax) [inline src/Main.cpp]: See above

File: src/Main.cpp [hunk 4]
```diff
@@ -138,25 +141,18 @@ PFNGLCOMPRESSEDTEXIMAGE2DARBPROC GLX_CompressedTexImage2D = 0;
...
-        imLog("* %s", extList.left(sp).data());
-        extList = extList.mid(sp + 1);
-    }
-    */
+    s_display = SDL_CreateWindow("imaginaryMyst Alpha", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, winWidth,
+                                 winHeight, SDL_WINDOW_OPENGL);
+
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_FLAGS, SDL_GL_CONTEXT_DEBUG_FLAG);
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 4);
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 3);
```
Comment (zrax) [inline src/Main.cpp]: Why 4.3 out of curiosity?  Do you have specific features or extensions in mind?

File: src/Main.cpp [hunk 5]
```diff
@@ -166,13 +162,14 @@ PFNGLCOMPRESSEDTEXIMAGE2DARBPROC GLX_CompressedTexImage2D = 0;
     glViewport(0, 0, winWidth, winHeight);
     glMatrixMode(GL_PROJECTION);
     glLoadIdentity();
-    gluPerspective(45.0f, (float)winWidth/(float)winHeight, 0.1f, 10000.0f);
+    gluPerspective(45.0f, (float) winWidth / (float) winHeight, 0.1f, 10000.0f);
```
Comment (zrax) [inline src/Main.cpp]: See above

File: src/CMakeLists.txt [hunk 6]
```diff
@@ -44,7 +44,7 @@ source_group("Source Files" FILES ${imCommon_SOURCES})
 add_executable(imaginaryMyst WIN32 ${imCommon_HEADERS} ${imCommon_SOURCES})
 target_link_libraries(imaginaryMyst imAnim imScene imSound imSurface imVfs)
 target_link_libraries(imaginaryMyst squish)
-target_link_libraries(imaginaryMyst ${SDL_LIBRARY} ${OPENGL_LIBRARIES})
+target_link_libraries(imaginaryMyst ${SDL2_LIBRARIES} ${OPENGL_LIBRARIES})
```
Comment (zrax) [inline src/CMakeLists.txt]: This didn't work for me -- I had to use the CMake import target (`SDL2::SDL2`) to get it to link.

File: src/imCommon.h [hunk 7]
```diff
@@ -17,8 +17,8 @@
 #ifndef _IM_COMMON_H
 #define _IM_COMMON_H
 
-#include <SDL.h>
-#include <SDL_opengl.h>
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_opengl.h>
```
Comment (zrax) [inline src/imCommon.h:21]: This type of change is also needed in `imThread.h`

File: src/Main.cpp (hunk 4 above)
Comment (Lunanne) [reply src/Main.cpp]: https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glDebugMessageCallback.xhtml is included from 4.3, which in my experience is very useful. However I think this made it in by accident as the rest of the code doesn't work yet under  OpenGL4 so I'll remove it.

**Pull request merged**